### PR TITLE
polyval operator

### PR DESCRIPTION
**Pull request by cliffburdick**

### CODE REVIEW

File: include/matx/operators/polyval.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        __MATX_INLINE__ PolyvalOp(const Op &op, const Coeffs &coeffs) : op_(op), coeffs_(coeffs) {
+          MATX_STATIC_ASSERT_STR(coeffs.Rank() == 1, matxInvalidDim, "Coefficient must be rank 1");
+          MATX_STATIC_ASSERT_STR(op.Rank() == 1, matxInvalidDim, "Input operator must be rank 1");
+        };
+
+        __MATX_INLINE__ __MATX_DEVICE__ __MATX_HOST__ auto operator()(index_t idx) const
+        {
+          scalar_type ttl{0};
+          for (int c = 0; c < coeffs_.Size(0); c++) {
+            ttl += coeffs_(c) * std::pow(op_(idx), coeffs_.Size(0) - c - 1);
```
[the remark this answers is not in the document]
Comment (cliffburdick) [reply include/matx/operators/polyval.h]: Good idea. fixed.

File: include/matx/operators/polyval.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        {
+          // Horner's method for computing polynomial
+          scalar_type ttl{coeffs_(0)};
+          for(int i = 1; i < coeffs_.Size(0); i++) {
+              ttl = ttl * op_(idx) + coeffs_(i);
+          }
+
+          return ttl;
+        }
+
```
Comment (tbensonatl) [inline include/matx/operators/polyval.h:73]: Do we need Pre/PostRun here for statements like `(y = polyval(conv1d(x, filter), coefs))`?

Comment (cliffburdick) [reply include/matx/operators/polyval.h:73]: Adding that and re-running tests.

**Review by tbensonatl — APPROVED** [review]

**Pull request merged**